pkg/registry/csv.go

```diff
@@ -53,6 +62,50 @@ type ClusterServiceVersion struct {
 	Spec json.RawMessage `json:"spec"`
 }
 
+// ReadCSVFromBundleDirectory tries to parse every YAML file in the directory and see if they are CSV.
+// According to the strict one CSV rule for every bundle, we return the first file that is considered a CSV type.
+func ReadCSVFromBundleDirectory(bundleDir string) (*ClusterServiceVersion, error) {
+	files, err := ioutil.ReadDir(bundleDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading bundle directory %s, %v", bundleDir, err)
+	}
+
+	ymlFiles := []string{}
+	for _, f := range files {
+		if !f.IsDir() {
+			ymlFiles = append(ymlFiles, f.Name())
+		}
+	}
+
+	for _, yamlFile := range ymlFiles {
+		csvReader, err := os.Open(path.Join(bundleDir, yamlFile))
+		if err != nil {
+			continue
+		}
+
+		unstructuredCSV := unstructured.Unstructured{}
+		csvStruct := ClusterServiceVersion{}
+
+		decoder := yaml.NewYAMLOrJSONDecoder(csvReader, 30)
+		if err = decoder.Decode(&unstructuredCSV); err != nil {
+			return nil, fmt.Errorf("error unmarshalling CSV, %v", err)
```

Comment:
but won't this return an error if the directory contains yaml files that aren't csvs?

Reply:
I will name this better. This just unmarshals to unstructured. It then tries to determin if it is CSV next 
` 		if unstructuredCSV.GetKind() != operators.ClusterServiceVersionKind {`